inception/inception-api-annotation-experimental/src/main/java/de/tudarmstadt/ukp/inception/experimental/api/websocket/AnnotationProcessAPIImpl.java

```diff
@@ -17,176 +17,296 @@
+    private static final String SERVER_RECEIVE_DELETE_ARC = "/arc_delete";
 
-    private static final String SERVER_RECEIVE_CLIENT_SELECTED_ANNOTATION = "/select_annotation_by_client";
-    private static final String SERVER_SEND_CLIENT_SELECTED_ANNOTATION = "/queue/selected_annotation_for_client/";
+    private static final String SERVER_SEND_DELETE_SPAN = "/topic/span_delete/";
+    private static final String SERVER_SEND_DELETE_ARC = "/topic/arc_delete/";
 
-    private static final String SERVER_RECEIVE_CLIENT_NEW_ANNOTATION = "/new_annotation_by_client";
-    private static final String SERVER_RECEIVE_CLIENT_DELETE_ANNOTATION = "/delete_annotation_by_client";
+    // UPDATE
+    private static final String SERVER_RECEIVE_UPDATE_SPAN = "/span_update";
+    private static final String SERVER_RECEIVE_UPDATE_ARC = "/arc_update";
 
-    private static final String SERVER_SEND_CLIENT_UPDATE_ANNOTATION = "/topic/annotation_update_for_clients/";
+    private static final String SERVER_SEND_UPDATE_SPAN = "/topic/span_update/";
+    private static final String SERVER_SEND_UPDATE_ARC = "/topic/arc_update/";
 
+    // ERROR
     private static final String SERVER_SEND_CLIENT_ERROR_MESSAGE = "/queue/error_message/";
 
     public AnnotationProcessAPIImpl(ProjectService aProjectService,
             DocumentService aDocumentService, UserDao aUserDao,
             RepositoryProperties aRepositoryProperties,
             SimpMessagingTemplate aSimpMessagingTemplate,
             AnnotationSchemaService aAnnotationSchemaService,
-            AnnotationSystemAPIService aAnnotationSystemAPIService)
+            ColoringService aColoringService, UserPreferencesService aUserPreferencesService)
     {
         this.simpMessagingTemplate = aSimpMessagingTemplate;
         this.annotationSystemAPIImpl = new AnnotationSystemAPIImpl(aProjectService,
                 aDocumentService, aUserDao, aRepositoryProperties, this, aAnnotationSchemaService,
-                aAnnotationSystemAPIService);
-    }
-
-    // ----------- ERROR HANDLER ------------- //
-    @MessageExceptionHandler
-    @SendToUser("/queue/errors")
-    public String handleException(Throwable aException)
-    {
-        return aException.getMessage();
+                aColoringService, aUserPreferencesService);
     }
 
     /**
      * ------------- PUB / SUB HANDLING -------------
      **/
 
     @Override
-    @MessageMapping(SERVER_RECEIVE_CLIENT_NEW_DOCUMENT)
-    public void handleReceiveDocumentRequest(Message<String> aMessage) throws IOException
+    @MessageMapping(SERVER_RECEIVE_DOCUMENT_REQUEST)
+    public void receiveDocumentRequest(Message<String> aMessage) throws IOException
     {
-        System.out.println("RECEIVED NEW DOCUMENT BY CLIENT, Message: " + aMessage);
-        annotationSystemAPIImpl.handleDocument(
-                JSONUtil.fromJsonString(ClientMessage.class, aMessage.getPayload()));
+        LOG.debug("RECEIVED DOCUMENT REQUEST BY CLIENT, Message: " + aMessage);
```

Comment:
`LOG.debug("RECEIVED DOCUMENT REQUEST BY CLIENT, Message: {}", aMessage);`

Do not use string concatenation when using SLF4J - use placeholders.